### DISCUSSION TITLE
Proposal: Constructor callable methods to initialize readonly fields

### DISCUSSION THREAD

**Discussion by phorks** [dotnet/csharplang#1296] [General]
It would be nice to be a way to declare a member method CustructorCallable so that it could only get called from a constructor and hence initialize readonly fields. This way each CunstructorCallable method may be responsible to initialize specific kind of readonly fields and since this method can be overridden, each subclass can use it to initialize its own readonly fileds in that specific category.

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/348

**Comment by MkazemAkhgary** [comment]
using `this` and `base` constructor calls, you can reduce most duplicate codes for initialization part. only in few cases you are forced to write duplicate code (if you want to keep things readonly). I don't think it is feasible to introduce new syntax for it which is proposed in #348 . instead an attribute seems better alternative which is proposed here.
